src/styles/breakpoints.js

```diff
@@ -43,5 +51,22 @@ export default function createBreakpoints(
     return breakpoints[name];
   }
 
-  return { keys, values, up, down, only, getWidth };
+  let matched = priority
+    .reduce((a, k) => {
+      const media = window.matchMedia(breakpoints[k]);
```

Comment:
Well I haven't fully thought about it. But one way is to add some listeners on `componentWillMount` and `setState`. Currently I've been using redux to monitor media query changes.